### PR TITLE
Update the nginx Docker image.

### PR DESCRIPTION
**Pull request by sepehre**
1. Install cron so that log rotation can work.
2. Optionally accept Service Name and Service Version as parameters for
Endpoints Server Proxy.

### CODE REVIEW

File: nginx_proxy/start_nginx.sh [hunk 1]
```diff
@@ -18,6 +18,43 @@ KEY_FILE=${CERT_DIR}/lb.key
...
+    -h
+        Shows this message.
+    -n ENDPOINTS_SERVICE_NAME
+        Required. The name of the Endpoints Service.
+        e.g. my-service.my-project-id.appspot.com
+    -v ENDPOINTS_SERVICE_VERSION
+        Required. The version of the Endpoints Service which is assigned
+        when deploying the service API specification.
+        e.g. 2016-04-20R662
+dd    -
```
Comment (oierw) [inline nginx_proxy/start_nginx.sh]: I think this may be some stray characters.

File: nginx_proxy/start_nginx.sh [hunk 2]
```diff
@@ -18,6 +18,43 @@ KEY_FILE=${CERT_DIR}/lb.key
...
+  cat << END_USAGE
+Usage: $(command basename $0) [options]
+Examples:
+(1) Starts nginx with no Endpoints service name and service version. In this
+mode either Endpoints features are disabled via nginx.conf or the service
+configuration is already provided on disk:
+$(command basename $0)
+(2) Starts nginx with a custom Endpoints service name and service version, and
+have nginx obtain the service configuration:
+$(command basename $0) -n SERVICE_NAME -v SERVICE_VERSION
```
Comment (oierw) [inline nginx_proxy/start_nginx.sh]: nit: Should probably have ENDPOINTS_ added to the variables.

File: nginx_proxy/start_nginx.sh [hunk 3]
```diff
@@ -42,4 +79,11 @@ if [[ -f "${CONF_FILE}" ]]; then
   cp "${CONF_FILE}" /etc/nginx/nginx.conf
 fi
 
+# fetch Service Configuration from Service Management if the service name and
+# service version are provided.
+if [[ -n "${SERVICE_NAME}" && -n "${SERVICE_VERSION}" ]]; then
+  /usr/sbin/fetch_service_config.sh \
+    -s "${SERVICE_NAME}" -v "${SERVICE_VERSION}" || exit $?
+fi
```
Comment (oierw) [inline nginx_proxy/start_nginx.sh]: Maybe print an error message and return non-zero if only one of -n or -v is provided?

File: nginx_proxy/start_nginx.sh [hunk 4]
```diff
@@ -42,4 +79,11 @@ if [[ -f "${CONF_FILE}" ]]; then
...
+
 /usr/sbin/nginx -p /usr -c /etc/nginx/nginx.conf
```
Comment (oierw) [inline nginx_proxy/start_nginx.sh]: Should this exec nginx (so we don't have a shell just lying around)?

File: nginx_proxy/start_nginx.sh (hunk 1 above)
Comment (sepehre) [inline nginx_proxy/start_nginx.sh]: Done.

File: nginx_proxy/start_nginx.sh (hunk 2 above)
Comment (sepehre) [inline nginx_proxy/start_nginx.sh]: Done.

**Comment by oierw** [comment]
LGTM

File: nginx_proxy/start_nginx.sh (hunk 3 above)
Comment (sepehre) [inline nginx_proxy/start_nginx.sh]: This functionality will move into the nginx.conf generator, so I'd rather not make it over-complicated in start_nginx.sh
Would you be ok with the current all or nothing approach?

File: nginx_proxy/start_nginx.sh (hunk 4 above)
Comment (sepehre) [inline nginx_proxy/start_nginx.sh]: I'm not sure, but it is out of scope for this review right?

**Comment by apolito** [comment]
lgtm

**Pull request merged**